### PR TITLE
ci: add docker build and push ci

### LINKED ISSUE #18: Add docker/build-push-action in the CI.
And replace the image repository [tianshimoyi](https://github.com/RocksLabs/kvrocks-operator/blob/unstable/deploy/operator/values.yaml#L2) with rockslabs

### PR DESCRIPTION
**Pull request by xiao-jay**
fix #18 
You shoud set github secret 'DOCKER_PASSWORD' and 'DOCKER_PASSWORD' first ,I have success build in local ci
https://github.com/xiao-jay/kvrocks-operator/actions/runs/4976932713
image

### CODE REVIEW

File: .github/workflows/docker-build-push.yaml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+      - name: build and publish image
+        env:
+          # fork friendly ^^
+          DOCKER_REPO: docker.io/${{ secrets.docker_username }}/kvrocks-operator
+        run: |
+          docker buildx build \
+          --platform linux/amd64,linux/arm64 \
+          --label "org.opencontainers.image.source=https://github.com/${{ github.repository_owner }}/kvrocks-operator" \
+          --label "org.opencontainers.image.description=kvrocks-operator image" \
+          --label "org.opencontainers.image.licenses=MIT" \
```
Comment (ColinChamber) [inline .github/workflows/docker-build-push.yaml]: We use the Apache-2.0 license

File: Dockerfile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-ARG USE_VENDOR=Disabled
+ARG USE_VENDOR=disabled
 
 # Build the manager binary
 FROM golang:1.17 as builder
```
Comment (ColinChamber) [inline Dockerfile]: Upgrade the version to 1.19

File: .github/workflows/docker-build-push.yaml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+  push:
+    branches:
+      - 'unstable'
+
+jobs:
+  docker-build-push:
+    strategy:
+      matrix:
+        go-version: [ 1.17.x ]
```
Comment (xiao-jay) [inline .github/workflows/docker-build-push.yaml]: @ColinChamber Do we need to update the Go version to 1.19？Also about the go version in docker file ,should update from 1.17 to 1.19?

Comment (ColinChamber) [reply .github/workflows/docker-build-push.yaml]: Yeah, you need to modify the version of golang used to build the image, [here](https://github.com/xiao-jay/kvrocks-operator/blob/build-image-ci/Dockerfile#L4).  Does docker runtime environment require a specific version of golang?  I suppose that the declaration of the go-version seems unnecessary here.

**Review by ColinChamber — CHANGES_REQUESTED** [review]

File: .github/workflows/docker-build-push.yaml (hunk 3 above)
Comment (xiao-jay) [reply .github/workflows/docker-build-push.yaml]: done

File: .github/workflows/docker-build-push.yaml (hunk 1 above)
Comment (xiao-jay) [reply .github/workflows/docker-build-push.yaml]: done

File: Dockerfile (hunk 2 above)
Comment (xiao-jay) [reply Dockerfile]: done

**Comment by xiao-jay** [comment]
@ColinChamber please review again

File: Dockerfile [hunk 4]
```diff
@@ -17,12 +17,12 @@ COPY pkg/ pkg/
...
 
 
-FROM builder as vendor-Enabled
+FROM builder as vendor-enabled
 ARG TARGETARCH
 # Build
 RUN CGO_ENABLED=0 GOOS=linux GOARCH=${TARGETARCH} go build -mod=vendor -a -o manager main.go
 
-FROM builder as vendor-Disabled
+FROM builder as vendor-disabled
```
Comment (ColinChamber) [inline Dockerfile:25]: [These](https://github.com/RocksLabs/kvrocks-operator/blob/unstable/Makefile#L132) should also be modified

**Comment by ColinChamber** [comment]
Suggest adding a check-docker job in CI.
```
  check-docker:
    name: Check Docker image
    needs: [lint-build]
    runs-on: ubuntu-20.04
    steps:
      - uses: actions/checkout@v3
      - uses: docker/build-push-action@v3
        with:
          context: .
          push: false
```

File: Dockerfile (hunk 4 above)
Comment (xiao-jay) [reply Dockerfile:25]: done

**Review by ColinChamber — APPROVED** [review]

File: .github/workflows/docker-build-push.yaml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+      - name: Login to  docker hub
+        uses: docker/login-action@v2
+        with:
+          registry: docker.io
+          username: ${{ secrets.DOCKER_USERNAME }}
+          password: ${{ secrets.DOCKER_PASSWORD }}
+      - name: build and publish image
+        env:
+          # fork friendly ^^
+          DOCKER_REPO: docker.io/${{ secrets.docker_username }}/kvrocks-operator
```
Comment (ColinChamber) [inline .github/workflows/docker-build-push.yaml]: Since secret names are case-insensitive, it seems better to use uppercase uniformly.

**Comment by xiao-jay** [comment]
> Suggest adding a check-docker job in CI.
> 
> ```
>   check-docker:
>     name: Check Docker image
>     needs: [lint-build]
>     runs-on: ubuntu-20.04
>     steps:
>       - uses: actions/checkout@v3
>       - uses: docker/build-push-action@v3
>         with:
>           context: .
>           push: false
> ```

May be one pr doing one thing is better ,I will do it in next pr.

File: .github/workflows/docker-build-push.yaml (hunk 5 above)
Comment (xiao-jay) [reply .github/workflows/docker-build-push.yaml]: sorry,I forgot to modify it because lowercase works fine too. I will change it right now.

Comment (xiao-jay) [reply .github/workflows/docker-build-push.yaml]: done

**Comment by ColinChamber** [comment]
Can I have your email? If you're willing, I would like to invite you to join some WeChat groups for kvrocks communities. @xiao-jay

**Comment by xiao-jay** [comment]
> 

sure，You can send a email to 895006375@qq.com,I will tell you my wechat.it's my honour.

**Pull request merged**